### ISSUE TITLE
Improve custom cross-entropy operator example

### ISSUE THREAD

**Issue by eric-haibin-lin** [apache/incubator-mxnet#8583]
In the recommender example, the custom cross-entropy operator can be implemented using NDArray instead of numpy, which gives much better performance. Changing the implementation to use NDArray will help other users who is learning MXNet / trying to use MXNet with custom ops.

https://github.com/apache/incubator-mxnet/blob/master/example/recommenders/crossentropy.py

**Comment by solin319** [comment]
I pull a new request in #8588.
The test result were equal between use np.array and mx.NDArray. The first test result are as below.
```
use np.array

D:[[ 0.69253981  0.29041654  0.23231442  0.5502885 ]
 [ 0.71507961  0.42464432  0.97114891  0.68113315]
 [ 0.48131326  0.39427516  0.34631446  0.72446871]
 [ 0.4398008   0.06848434  0.40008336  0.73323548]
 [ 0.18884189  0.18194272  0.53092033  0.53119105]
 [ 0.63171291  0.84244317  0.71996623  0.60880303]]
L:[[ 0.  0.  1.  1.]
 [ 0.  0.  1.  0.]
 [ 1.  0.  0.  1.]
 [ 0.  1.  0.  1.]
 [ 0.  0.  0.  0.]
 [ 1.  0.  0.  1.]]
out:[[-1.17940652 -0.34307581 -1.45965922 -0.59731078]
 [-1.25554216 -0.55276525 -0.02927442 -1.14297867]
 [-0.73123485 -0.50132793 -0.42512736 -0.32231531]
 [-0.57946116 -2.68113565 -0.51096296 -0.310287  ]
 [-0.2092911  -0.20082173 -0.7569806  -0.75755793]
 [-0.45931864 -1.84796298 -1.27284169 -0.49625883]]

use mx.nd.array

D:[[ 0.69253981  0.29041654  0.23231442  0.5502885 ]
 [ 0.71507961  0.42464432  0.97114891  0.68113315]
 [ 0.48131326  0.39427516  0.34631446  0.72446871]
 [ 0.4398008   0.06848434  0.40008336  0.73323548]
 [ 0.18884189  0.18194272  0.53092033  0.53119105]
 [ 0.63171291  0.84244317  0.71996623  0.60880303]]
L:[[ 0.  0.  1.  1.]
 [ 0.  0.  1.  0.]
 [ 1.  0.  0.  1.]
 [ 0.  1.  0.  1.]
 [ 0.  0.  0.  0.]
 [ 1.  0.  0.  1.]]
out:[[-1.17940652 -0.34307581 -1.45965922 -0.59731078]
 [-1.25554216 -0.55276525 -0.02927442 -1.14297867]
 [-0.73123485 -0.50132793 -0.42512736 -0.32231531]
 [-0.57946116 -2.68113565 -0.51096296 -0.310287  ]
 [-0.2092911  -0.20082173 -0.7569806  -0.75755793]
 [-0.45931864 -1.84796298 -1.27284169 -0.49625883]]
```

**Comment by zihaolucky** [comment]
Add a performance test against NDArry and Numpy array would be great.

**Comment by eric-haibin-lin** [comment]
I found that the actual loss calculate is missing a negative sign in this example: https://github.com/apache/incubator-mxnet/blob/master/example/recommenders/crossentropy.py#L57 
This should be fixed

**Comment by ThomasDelteil** [comment]
@eric-haibin-lin it looks like the example has been updated to use ndarray rather than numpy. Can you please close?
